### PR TITLE
[4] Remove method from exception messages

### PR DESCRIPTION
**Pull request by PhilETaylor**
### Summary of Changes

Continuing the slow progress of removing PHP Internal Class and Method names from user visible error messages. 

Including class/method names and namespaces is "probably" ok for framework packages aimed at developers - but class/method names should never be shown to end users in error messages. (E.g When was the last time an Amazon shopping error page told you which class and method the error was in? it never does!) 

### Testing Instructions

mainly code review. 

https://example.com/administrator/index.php?option=com_media&format=json&task=api.files&path=local-images:/sampledata/../banner_cradle.jpg&url=asddfsdfg

### Actual result BEFORE applying this Pull Request

Screenshot 2021-05-23 at 14 43 31

### Expected result AFTER applying this Pull Request

Screenshot 2021-05-23 at 14 48 02

### CODE REVIEW

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 98fded7dea95ba0d44d3e060934764542eed4554<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34130">issues.joomla.org/tracker/joomla-cms/34130</a>.</sub>

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 98fded7dea95ba0d44d3e060934764542eed4554<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34130">issues.joomla.org/tracker/joomla-cms/34130</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34130">issues.joomla.org/tracker/joomla-cms/34130</a>.</sub>

File: libraries/src/Filesystem/Path.php [hunk 1]
```diff
@@ -162,34 +163,22 @@ public static function getPermissions($path)
...
-			// Don't translate
-			throw new \Exception(
-				sprintf(
-					'%s() - Use of relative paths not permitted',
-					__METHOD__
-				),
-				20
-			);
+			// Don't translate.
+			throw new Exception('Use of relative paths not permitted', 20);
```
Comment (wilsonge) [inline libraries/src/Filesystem/Path.php]: Can we use this opportunity to remove the status code of 20 as well on these exceptions. These also date back to JError implementations.

**Comment by richard67** [comment]
Back to pending due to updates requested.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34130">issues.joomla.org/tracker/joomla-cms/34130</a>.</sub>

**Comment by PhilETaylor** [comment]
Sorry for the delay. Life. 

Code 20 is now removed.

File: libraries/src/Filesystem/Path.php (hunk 1 above)
Comment (PhilETaylor) [reply libraries/src/Filesystem/Path.php]: Done.

**Comment by PhilETaylor** [comment]
Conflicts resolved.

**Pull request closed without merging**